src/sentry/sentry_metrics/indexer/postgres_v2.py

```diff
@@ -17,8 +19,21 @@
 # only used to compare to the older version of the PGIndexer
 _INDEXER_CACHE_FETCH_METRIC = "sentry_metrics.indexer.memcache.fetch"
 
+DEFAULT_INDEXER_TYPE = "release-health"
+
+IndexerTable = TypeVar("IndexerTable", bound=BaseIndexer)
+
+TABLE_MAPPING = {
+    "release-health": StringIndexerTable,
+    "performance": PerfStringIndexer,
+}
+
 
 class PGStringIndexerV2(StringIndexer):
+    def __init__(self):
+        ingest_profile = get_ingest_config()
```

Comment:
Coming from a more recent Java background, this sort of seems to me like it should be wired in, but it didn't really make it cleaner in my opinion to put the setting in server.py and have it pass through the `StaticStringsDecorator` and through to here. 

If people want it that way I can make the change, though.

Reply:
I think the issue you are running into is that a service (LazyServiceWrapper) is supposed to exist in one copy per deplyoment. Obviously each node/pod would have its instance but they will all be configured in the same way for the same deployment (saas, open source, single tenant, etc.). 

There is a concept called `ServiceDelegator` https://github.com/getsentry/sentry/blob/fbe987561d2b9df070f34652785294cc430b41e4/src/sentry/utils/services.py#L419-L449 that allows you to configure multiple backends for a service with different options and allows you to specify a selector function that picks up the service.

As long as the indexer is a `Service`, if we need multiple different implementations of it coexisting I fear we will have to run multiple ones via the service delegator above. This is because  the indexer can pick one by environment variable as we do not change it during the lifetime of a pod. Sentry backend instead is a monolith, a pod will have to be able to access both at the same time depending what api endpoint the user is hitting. So we cannot select it by environment variable. 

The ServiceDelegator allows you to look into the call parameters of each method to select the appropriate backend, which means you will have to add the use case id in each method signature. This pollutes a bit the interface for no reason. But at least it would allow the Sentry api to work with multiple indexers.

Another alternative is declaring another service for this new indexer with the same interface, but I am not sure if this even works. 

 I think using the delegator may not be that bad of an idea in the end. `use_case_id` can be seen as an abstract partition of the indexer. Who uses it does not need to care whether there is actually one or 1000 separate indexer instances. They just care about what use case they are dealing with in abstract.

Reply:
I wonder if we can do something even simpler than this, if we have to add `use_case_id` to all the method signatures anyway. The only two things we're changing in the `StringIndexer` service itself are:
1. what table do we read from?
2. how do we tag the metrics?

For both of these, we can dynamically per-call use the correct table and the correct metrics tag without branching between two instances of the service class e.g.

```python
def _table(use_case_id: ProfileKey):
   return TABLE_MAPPING[get_config_for_profile(use_case_id)]

def _pipeline_tag(use_case_id: ProfileKey):
  return get_config_for_profile(use_case_id).internal_metrics_tag
```

and get rid of most of the dependency on `os.environ` altogether. For switching the Kafka topics, we could take a profile name from the CLI arguments.

What do you think?

Reply:
Works for me.